woocommerce/woocommerce-gutenberg-products-block
# Bump woocommerce/woocommerce-sniffs from 0.1.0 to 0.1.3

Fixes woocommerce/woocommerce-blocks#7657 

Update the collection of [PHP_CodeSniffer sniffs for WooCommerce](https://github.com/woocommerce/woocommerce-sniffs) to the latest version 0.1.3.

This new release adds validation for the [at]since version for all hooks.

## Summary of the changes:
- The codebase was updated to account for the newly introduced `@since` tag validation. 
- Added Docblocks to all hooks that didn't have one yet.
- The hooks documentation was also updated (via `npm run build:docs`).

## Note

A few hooks (the older ones) didn't have a changelog or a milestone associated with them: for these cases, the release closest to the date the PR was merged to `trunk` was used as the reference.

## Discussion

Considering that with this update, the usage of the `@since` tag is now part of our validation, we might want to update our [release process](https://github.com/woocommerce/woocommerce-blocks/blob/trunk/.github/release-initial-checklist.md) accordingly: e.g. it is a common practice to:
- Add a `@since TBD` on the docblock when creating a new hook.
- Search and replace all occurrences of `TBD` with the correct version number when it is time to release. This could be a new task for the release manager, and it is possible to automate it. 
- Some of the hooks (e.g. `'woocommerce_email_header'`) are native to WooCommerce or the WP core, and we are re-purposing them here on `woocommerce-blocks`: for the sake of consistency, the `@since` version for those correspond to when they were added to this repo. We could also make an exception, not add a version for those at all, and reference the original hook instead. Open to hearing others' thoughts.

Leaving this open for discussion in case folks have any other suggestions.

### Testing



The `PHP Coding Standards / PHP coding standards` check should be successful on this PR. 

To test locally:

- `composer install`
- Run the sniffs locally: e.g. via command line: `./vendor/bin/phpcs --standard=WooCommerce-Core <file>`.
- Ensure everything works as expected: You should not see any errors related to missing the `@since version` for hooks. You should not see any errors related to missing docblocks for hooks.

### Changelog

> Added the [at]since version for all hooks. Added docblocks for hooks that didn't have one yet.
